pdr_backend/predictoor/base_predictoor_agent.py

```diff
@@ -82,22 +89,90 @@ def take_step(self):
         submit_epoch, target_slot = self.cur_epoch, self.target_slot
         logger.info("Predict for time slot = %s...", self.target_slot)
 
-        predval, stake = self.get_prediction(target_slot)
-        logger.info("-> Predict result: predval=%s, stake=%s", predval, stake)
-        if predval is None or stake <= 0:
-            logger.warning("Done: can't use predval/stake")
+        success = self.get_prediction_and_submit(target_slot)
+        if not success:
             return
```

Comment:
This is more manageable complexity than @trizin version, because it doesn't added 50+ more lines to `take_step()` to handle both one-sided and two-sided prediction. Rather, it pushes that functionality into a sub-method.